cli/flox/src/commands/activate.rs

```diff
@@ -275,17 +275,18 @@ impl Activate {
 
             // TODO: we should clean up the different conditionals here
             if in_place && self.start_services {
+                debug!("not starting services for in-place activation");
                 message::warning("Skipped starting services. Services are not yet supported for in place activations.");
             }
             if flox.features.services && !manifest.services.is_empty() && !in_place {
                 if self.start_services {
                     supported_environment(&flox, self.environment)?; // Error for remote envs.
                 }
                 tracing::debug!(start = self.start_services, "setting service variables");
-                if socket_path.exists() {
+                if self.start_services && socket_path.exists() {
                     debug!("detected existing services socket");
                     message::warning("Skipped starting services, services are already running");
-                } else {
+                } else if self.start_services {
```

Comment:
(blocking) I think this part was correct previously but the warning message when the socket already exists was wrong. If we don't set the environment variable based on the current activation then we'll leak a positive value down to nested activations that haven't specified `--start-services`, similar to the problem in https://github.com/flox/flox/pull/1895:
```suggestion
                } else {
```

(both cases probably warrant tests but we could do that in a separate PR)

Reply:
Actually, the suggestion by itself might not be the right logic, but you get the idea.

Reply:
I'm not sure I follow. Do we just need to set it regardless but set it to a negative value when appropriate?

Reply:
Yep 👍🏻

Reply:
Fixed